src/mococrw/util.h

```diff
@@ -34,5 +38,43 @@ std::string toHex(const std::vector<uint8_t> &data);
 std::vector<uint8_t> fromHex(const std::string &hexData);
 
 std::vector<uint8_t> cryptoRandomBytes(size_t length);
+
+template<typename T>
+void vectorCleanse(std::vector<T> &vec)
+{
+    OPENSSL_cleanse(vec.data(), vec.size() * sizeof(T));
+}
+
+/**
+ * RAII wrapper that executes a lambda on destruction
+ */
+class Finally {
+public:
+    /**
+     * Create a new finally object, which will run the given function when it
+     * goes out of scope.
+     *
+     * @param[in] func Function to run on destruction.
+     */
+    Finally(std::function<void()> func)
```

Comment:
since this is now a general type here:

This should take a universal reference to arbitrary things and std::forward it to the std::function.

Reply:
Done